### PR TITLE
docs: Add example for `AzureSynapseRunPipelineOperator`

### LINKED ISSUE #36488: Missing AzureSynapseRunPipelineOperator in docs
### What do you see as an issue?

There are two operators in the synapse.py file (https://github.com/apache/airflow/blob/main/airflow/providers/microsoft/azure/operators/synapse.py), but on the page https://airflow.apache.org/docs/apache-airflow-providers-microsoft-azure/stable/operators/azure_synapse.html AzureSynapseRunPipelineOperator is not described (missing)

### Solving the problem

_No response_

### Anything else

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by Lymah123**
This PR makes changes to the documentation on the AzureSynapes

Closes: #36488 

**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by Lymah123** [comment]
Hi @rozza , @imbaczek , @jonshea , @dacort . Please I will be waiting for your reviews.

File: docs/apache-airflow-providers-microsoft-azure/operators/azure_synapse.rst
Comment (eladkal) [inline docs/apache-airflow-providers-microsoft-azure/operators/azure_synapse.rst]: This reference is wrong here as it's of `AzureSynapseRunSparkBatchOperator`.
If we don't have example for `AzureSynapseRunPipelineOperator` then you need to create one give it a new reference and then you can utalize it here.

For example:

https://github.com/apache/airflow/blob/63cc915cd38a5034df6bf9c618e12f8690eeade0/tests/system/providers/microsoft/azure/example_azure_synapse.py#L64-L70

Comment (Lymah123) [reply docs/apache-airflow-providers-microsoft-azure/operators/azure_synapse.rst]: Airflow code snippet

I saw this @eladkal for `AzureSynapseRunPipelineOperator`. Can I reference this?

Comment (eladkal) [reply docs/apache-airflow-providers-microsoft-azure/operators/azure_synapse.rst]: Yes. you will need to add `[START howto_operator_azure_synapse_run_pipeline]`  /  `[END howto_operator_azure_synapse_run_pipeline]` then you can reference it. Check the link I shared above

File: tests/system/providers/microsoft/azure/example_synapse_run_pipeline.py
Comment (eladkal) [inline tests/system/providers/microsoft/azure/example_synapse_run_pipeline.py]: Why do we need a new `AzureSynapseRunPipelineOperator`?
Can we wrap  `run_pipeline1` with the `howto_operator_azure_synapse_run_pipeline` tags?

Comment (Lymah123) [reply tests/system/providers/microsoft/azure/example_synapse_run_pipeline.py]: Why do we need a new AzureSynapseRunPipelineOperator?

It gave me an error while making the changes in VS code.

Can we wrap run_pipeline1 with the howto_operator_azure_synapse_run_pipeline tags?
I will change that now.

Comment (Lymah123) [reply tests/system/providers/microsoft/azure/example_synapse_run_pipeline.py]: @eladkal , do you mean I should replace task_id with this `howto_operator_azure_synapse_run_pipeline`?
Sorry, I didn't that.

Comment (eladkal) [reply tests/system/providers/microsoft/azure/example_synapse_run_pipeline.py]: My suggestion is:

```diff
) as dag:
-    # [START howto_operator_azure_synapse_run_pipeline]
-    run_spark_job = AzureSynapseRunPipelineOperator(
-        task_id="run_pipeline_job",
-        spark_pool="provpipelinepool",
-        payload=PIPELINE_JOB_PAYLOAD,  # type: ignore
-    )
-    # [END howto_operator_azure_synapse_run_pipeline]
    

    begin = EmptyOperator(task_id="begin")
+    # [START howto_operator_azure_synapse_run_pipeline]
    run_pipeline1 = AzureSynapseRunPipelineOperator(
        task_id="run_pipeline1",
        azure_synapse_conn_id="azure_synapse_connection",
        pipeline_name="Pipeline 1",
        azure_synapse_workspace_dev_endpoint="azure_synapse_workspace_dev_endpoint",
    )
+    # [END howto_operator_azure_synapse_run_pipeline]
    begin >> run_pipeline1
```

Comment (Lymah123) [reply tests/system/providers/microsoft/azure/example_synapse_run_pipeline.py]: OKAY! 

You can check again @eladkal ,

Comment (eladkal) [reply tests/system/providers/microsoft/azure/example_synapse_run_pipeline.py]: Let me explain 
the code block of `run_pipeline1` already exist

https://github.com/apache/airflow/blob/5983506df370325f7b23a182798341d17d091a32/tests/system/providers/microsoft/azure/example_synapse_run_pipeline.py#L41-L46

You just need to add the `howto_operator_azure_synapse_run_pipeline` around it.

If you will check your code you will see that you duplicated the block:

```
   begin = EmptyOperator(task_id="begin")
    # [START howto_operator_azure_synapse_run_pipeline]
    run_pipeline1 = AzureSynapseRunPipelineOperator(
        task_id="run_pipeline1",
        azure_synapse_conn_id="azure_synapse_connection",
        pipeline_name="Pipeline 1",
        azure_synapse_workspace_dev_endpoint="azure_synapse_workspace_dev_endpoint",
    )
    # [END howto_operator_azure_synapse_run_pipeline]
    begin >> run_pipeline1

    begin = EmptyOperator(task_id="begin")

    run_pipeline1 = AzureSynapseRunPipelineOperator(
        task_id="run_pipeline1",
        azure_synapse_conn_id="azure_synapse_connection",
        pipeline_name="Pipeline 1",
        azure_synapse_workspace_dev_endpoint="azure_synapse_workspace_dev_endpoint",
    )
```

**Comment by Lymah123** [comment]
Hi @eladkal , I have seen the duplicate you're talking about.  
You can check again.  
Thanks for being patient with me.

File: tests/system/providers/microsoft/azure/example_synapse_run_pipeline.py
Comment (eladkal) [inline tests/system/providers/microsoft/azure/example_synapse_run_pipeline.py]:
```suggestion
    begin = EmptyOperator(task_id="begin")
```

You changed the indent of this line thus it causes tests to fail.
This will fix the problem

File: docs/apache-airflow-providers-microsoft-azure/operators/azure_synapse.rst
Comment (eladkal) [inline docs/apache-airflow-providers-microsoft-azure/operators/azure_synapse.rst]: This is not the right description for this operator. Probably a copy-paste mistake as you used `AzureSynapseRunSparkBatchOperator` for template.
```suggestion
The operator will Executes a Synapse Pipeline.
```

**Comment by Lymah123** [comment]
@eladkal , you can review it again. Thanks.

File: docs/apache-airflow-providers-microsoft-azure/operators/azure_synapse.rst
Comment (eladkal) [inline docs/apache-airflow-providers-microsoft-azure/operators/azure_synapse.rst]:
```suggestion
      :start-after: [START howto_operator_azure_synapse_run_pipeline]
```

File: docs/apache-airflow-providers-microsoft-azure/operators/azure_synapse.rst
Comment (eladkal) [inline docs/apache-airflow-providers-microsoft-azure/operators/azure_synapse.rst]:
```suggestion
      :end-before: [END howto_operator_azure_synapse_run_pipeline]
```

**Review by eladkal — APPROVED** [review]
LGTM
Will merge when tests are green

**Comment by Lymah123** [comment]
> LGTM Will merge when tests are green



> LGTM Will merge when tests are green

Yaay! Thanks for being patient with me. @eladkal ! Like how you give feedback!

File: docs/apache-airflow-providers-microsoft-azure/operators/azure_synapse.rst
Comment (eladkal) [inline docs/apache-airflow-providers-microsoft-azure/operators/azure_synapse.rst]:
```suggestion
.. exampleinclude:: /../../tests/system/providers/microsoft/azure/example_synapse_run_pipeline.py
```

**Comment by eladkal** [comment]
congrats @Lymah123 on your first merged PR!

**Pull request merged**
